kaitai/struct.go

```diff
@@ -0,0 +1,12 @@
+package kaitai
+
+// Struct is the common interface guaranteed to be implemented by all types generated by
+// Kaitai Struct compiler.
+type Struct interface {
+	// Kaitai_IO returns the stream object associated with the struct.
+	//
+	// This is deliberately named with a `Kaitai_` prefix and underscore to avoid conflicts
+	// with other methods that may result from the attributes in Kaitai Struct type, e.g.
+	// is a user will define attribute `i_o` this will conflict with the method `IO()`.
+	Kaitai_IO() *Stream
```

Comment:
> `IO_()`

This is my new favorite. I knew that if you start the identifier with an underscore `_`, then it doesn't start with a capital letter and won't be exported, but I didn't think of _ending_ the identifier with an underscore `_`. It's simple and I suppose it potentially works better with autocompletion (assuming that only identifiers starting with what you type are shown) - in case a user would sometimes need to refer to `_io`, `_parent` or `_root`, they can start typing `I`, `Par` or `Ro` and will be offered `IO_()`, `Parent_()` and `Root_()` respectively, and they don't have to remember that there is a special `Kaitai_` prefix.

Reply:
Thanks! I also like the idea of `IO_()`. Just to get an Go expert opinion: how bad it will be from ecosystem perspective — e.g. it will generate code that will have to break varnames linting? How serious it is?

On the side note, to be honest, I won't mind breaking compat and turning Go `Read` method into `Read_` for example (given that vast majority of other languages generates `_read()` or some similar).